src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -2880,6 +2858,56 @@ namespace EvaporativeCoolers {
+        // Determines current operating mode of indirect research special evaporative cooler
+        // from the five valid operating modes depending the primary and secondary air
+        // temperatures, setpoint temperature, and full capacity air outlet temperature.
+
+        // METHODOLOGY EMPLOYED:
+        // compares various temperatures to determine the operating mode
+
+        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+        Real64 InletDryBulbTempPri; // entering air dry bulb temperature of primary air
+        Real64 SysTempSetPoint;     // evaporative cooler outlet setpoint temperature, drybulb
+        int OperatingMode;          // current operating mode of indrect evaporative cooler
+
+        InletDryBulbTempPri = EvapCond(EvapCoolNum).InletTemp;
+        SysTempSetPoint = EvapCond(EvapCoolNum).DesiredOutletTemp;
+
+        // Now determine the operating modes of indirect evaporative cooler research special. There are five allowed operating modes
+        if ((InletDryBulbTempPri <= SysTempSetPoint) ||
+            (InletDryBulbTempPri > EvapCond(EvapCoolNum).MaxOATDBEvapCooler && InletWetBulbTempSec > EvapCond(EvapCoolNum).MaxOATWBEvapCooler) ||
+            (InletDryBulbTempPri < InletDryBulbTempSec)) {
+            OperatingMode = None;
+        } else if ((InletDryBulbTempSec < EvapCond(EvapCoolNum).MinOATDBEvapCooler && TdbOutSysDryMin < SysTempSetPoint)) {
+            OperatingMode = DryModulated; // dry mode capacity modulated
+        } else if ((InletDryBulbTempSec < EvapCond(EvapCoolNum).MinOATDBEvapCooler && SysTempSetPoint <= TdbOutSysDryMin)) {
+            OperatingMode = DryFull; // dry mode in full capacity
+        } else if ((InletDryBulbTempSec >= EvapCond(EvapCoolNum).MinOATDBEvapCooler &&
+                    InletWetBulbTempSec < EvapCond(EvapCoolNum).MaxOATWBEvapCooler && SysTempSetPoint <= TdbOutSysWetMin)) {
+            OperatingMode = WetFull; // wet mode in full capacity
+        } else if ((InletDryBulbTempSec >= EvapCond(EvapCoolNum).MinOATDBEvapCooler &&
+                    InletWetBulbTempSec < EvapCond(EvapCoolNum).MaxOATWBEvapCooler &&
+                    TdbOutSysWetMin < SysTempSetPoint)) { // && SysTempSetPoint < TdbOutSysDryMin
+            OperatingMode = WetModulated;                 // wet mode capacity modulated
+        } else if ((InletDryBulbTempSec >= EvapCond(EvapCoolNum).MinOATDBEvapCooler &&
+                    InletDryBulbTempSec < EvapCond(EvapCoolNum).MaxOATDBEvapCooler &&
+                    InletWetBulbTempSec < EvapCond(EvapCoolNum).MaxOATWBEvapCooler && SysTempSetPoint < TdbOutSysDryMin &&
+                    TdbOutSysWetMin < SysTempSetPoint)) {
+            OperatingMode = DryWetModulated; // modulated in dry and wet mode, and the lower total power will be used
+        } else {
+            OperatingMode = None; // this condition should not happen unless the bounds do not cover all combinations possible
+        }
+        return OperatingMode;
+    }
```

Comment:
I like that you extracted this into a proper worker function with a singular purpose.  Easy to test and maintain going forward.